### ISSUE TITLE
Missing DateTime.SpecifyKind

### ISSUE THREAD

**Issue by darting** [fable-compiler/Fable#1844]
### Description

Missing DateTime.SpecifyKind method. 

Actually my problem is when I use Thoth Json to decode the datetime string `2019-06-03T09:08:31.2330000`, which is in UTC time, it will actually become my local time zone, and I am trying to convert it to UTC time, any idea?  Thanks

### Repro code

https://fable.io/repl/#?code=PYBwpgdgBAygngZwC5gLYChSSgIQE7ADuCYe66ANmElGBVABQJQBcUyeAlhAOYCUUALzoooqFRpghUACbAAxgFdUkJADp5eMAEMUAUSoqISBgCIZnAG6m+IsWDUoAHkgDCwY6qgAeALTs7UTByCShtEHAIGQBZBB5GZjYObn4hQPFqKBUEBG0eMGZBWQVlVTUAR0VSOBg6MHkkYDwzAGJs3PybdOYAHwA+Wnp+rIKOgrVwyJlXAAtOChkoYc4eCCbgykyZGiKAEV0wABVOFTUABW08EjMAJgAGAEYATl87gDZXgGZDu6eWO4AHCxPg81DdPp87lC7jY1IdgABVCBWUi5CjHFQMWzobZw4AwJBcXhYpYDSaQGJxHHqeEEok8MxwJlwNTRaJqGSLAASXJYqFQLByNlJYQiFNiPHImxo2xu0n2KAxDhg4HknAAZnAANLcGQMbYAGigCqOJzAOqiagRSHktlleLpKRJw3JUQl1JuDsJTtMzJZbI53N5-MFCGFLrFbqpQA&html=DwCwLgtgNgfAUHUBTAhgE3gAk8CSwqYDGIKATgM74C8ARAK5gBmAtABy3zAD0IqGCYACMA9mgCeWHGgCWAN0wy0dPBQooA5kk4BZJGs1JMABygoiSECKhokZHrLlduoifES9IsIA&css=Q

**Comment by alfonsogarciacaro** [comment]
Hmm, didn't know about that method but looks a bit dangerous. You can change between UTC and local time using the `.ToLocalTime()` and `.ToUniversalTime()` methods.

> In any case, we should throw a compilation error to warn SpecifyKind is not supported.

**Comment by rtkelly13** [comment]
In this case DateTime.SpecifyKind is necessary!

ToUniversalTime() will change the time underneath so 2019-06-03T09:08:31.2330000 will change to 2019-06-03T07:08:31.2330000 if in a +2 time zone. The issue here is the parsing cannot detect that the string given is in UTC and assumes that it is a local time. There should be a Z on the end for the string to parse into UTC. DateTime.Kind also has an Unspecified type which has some pitfalls when using those methods. 
https://docs.microsoft.com/en-us/dotnet/api/system.datetime.touniversaltime?view=netframework-4.8

I think the API design which mutates the DateTime was identified as dangerous which is why it isn't an instance method.

If only NodaTime was supported 😉

**Comment by alfonsogarciacaro** [comment]
Ah, thanks @rtkelly13! Didn't notice the `Z` was missing. In that case, the method may be indeed necessary though IMHO it would be better to deal with this during serialization. Don't remember exactly how it is in Thoth.Json but I think we tried to imitate the behavior in Newtonsoft.Json so it's possible it does default to local time if no time zone info is provided.

You can use an extra decoder to change this behavior, like this:

```fsharp
open System
open System.Text.RegularExpressions
open Thoth.Json

let private timeZoneRegex = Regex(@"([+-][0-2]\d:[0-5]\d|Z)$")

let datetimeSafe path (value: JsonValue) =
    let value =
        match value with
        | :? string as dt ->
            if timeZoneRegex.IsMatch dt then box dt                
            else box(dt + "Z") // Assume it's UTC
        | _ -> value
    Decode.datetime path value
``` 

👉 [Example in the REPL](https://fable.io/repl/#?code=PYBwpgdgBAygngZwC5gLYChSVolqB0AKmAB5L4BKYA5gK4A2AhgE4CiJIzYCCAlsBASZw0AELNgAdwRhmw7IQAWwJIvwApBAPnQAYowBG9MJTCMAxkh1RdDVI3TpUwACYMwUACJhzrjwF50KGCoYyQoTl4AN0YUKCReVDAALQEwKmpSKH8oDNIACgABACJ8gG0AagBaAF0ygAYqgCYagB0XAC4GqoBWNpcAH2SASgASYuHHENCwcJdY2cSwGEYAMw8QWMUofJj6WjAOqE0BADVGfbBh7KDpkLCoPYObu9eoeyRzbaePSV5VW5vEIDKAdAD8UGQzF4EGoUEYCCgLnCVQAfICgXdeKt4ktUhB0jRSPgAJIIACysS+SPCqmwBmAJBpmLeGJZwTA9BkUAZJHyyKgFSgxWSEygAHpxVAAII8WhJKD-ADkiIAqoQAMJst4ggD6UDRjwuB21wW8vhcJnmKASCs2qiNl0czjcxly3AY4UC0wetAgkmYjBA2Sgqz9ln4EFNUBBAHkANZQJmGkjRkGsZgSZhQWQG1Ghxi8eh-B35WREYAwJDQ2H5YaTJyudxQdjVxgvEKSqBKXiIpKqVyKxEQFTwmJFwxumHxRQeCisAAKABl4UhUAAaSHAKCSDbAehwVZF+iK8KzrgYh4w+gw37-RQa2jIYCoHaQC2yI6sCAf5gAHiVc5LlRa5+R8PxmCOc0IIAoCDhAt8yEDL8kMYDUIIQYYULbdDLWYRFvTeSkQHwRgXBceI4HAYBVlg40wFRfBbHoegADlGAVfJvw-fBeW4iCcx-CDN2gy1eMZUTc0tX9rlINtHAeOTA1w2QCIxYj8DQEAkDgDEBnzVtA3wEsHyfJAXxbISxOtRYFUk-AbNtZY1jABTZhpKCFkIJYO2CYpWmKJp6gARgATiqeoADYIoAZkIepQo6eoAA4Ohi4L8CaGKYvqXL6gC4o9PzMNoERQ17OlWhzPwVYJFQKsa2ofIEE3JTGH8NqVPwyZpn090EE9fA-QDINHEIScTCQCaoDKbsJtJBAAAlgCiWRppqGaMSQBkXDgGaNrKaNq32zb2RnE6yihYVPAWKBA0kYoamjaYkAosoDqu-lyEIStqxhJr6yeoEgbeY73tO9kHXBy7juKG64lWYBmA+FAXEe56Qlei7PuRCsGv+-JijgYm4HwclyQcijFsWjpUFQDoeAmYYQdeFmoCep6+ucP0kFEOASQo4okh4RhMkKoA&html=DwCwLgtgNgfAUHUBTAhgE3gAk8CSwqbhgAOAtEgI4CuAlgG4C8A5AMID2AdmEt2QCoBPEkmaYAxlx7cWPAB5gA9OGgBuCSBQAnAM75G1MADMyADmZYcO8VtokwmHVvGMARAH13ACQDyAZX5PRShaACMdRS1UcTAAOhItdjRqGNouWIhaTliAKx1XGGBFa1t7S2ASuwcnFw9vf0D3YLCIqJQYsjR2CHjE5NT0zOy8gqLKsrhsYBDOAGtMKKg3HTBBKCQdECR8VyIoozdPXwCgkPDFcR0I0OooCBQMrNjL-MxFeCKt9HhEUKTBcpoBiYWhoNx4K4oADmSAKAFkNjpoUhMCQoO0kCB2FA0EgtEUgfQPoo-mgAQhPpBYEA&css=Q)
